File: .repro/check_api.py
#!/usr/bin/env python3
import urllib.request
import urllib.error
import json
import ssl

CM_HOST = "https://ec2-44-205-177-172.compute-1.amazonaws.com"
KEY_ID = "f11dd172f3ee4632ad55bb4834bcb95a4a73384647474fa3993bbab4862c7849"

ctx = ssl.create_default_context()
ctx.check_hostname = False
ctx.verify_mode = ssl.CERT_NONE

# Step 1: get token
auth_data = json.dumps({
    "grant_type": "password",
    "username": "admin",
    "password": "Koala2.20",
    "domain": "root",
    "auth_domain": "root"
}).encode()

req = urllib.request.Request(
    f"{CM_HOST}/api/v1/auth/tokens",
    data=auth_data,
    headers={"Content-Type": "application/json"},
    method="POST"
)
with urllib.request.urlopen(req, context=ctx) as resp:
    token_data = json.load(resp)
token = token_data["jwt"]
print(f"Token obtained: {token[:20]}...")

# Step 2: query the key
req2 = urllib.request.Request(
    f"{CM_HOST}/api/v1/vault/keys2/{KEY_ID}",
    headers={"Authorization": f"Bearer {token}"},
    method="GET"
)
with urllib.request.urlopen(req2, context=ctx) as resp:
    key_data = json.load(resp)

print("\n=== CM API Response for key ===")
print(f"id                : {key_data.get('id', '<NOT SET>')}")
print(f"name              : {key_data.get('name', '<NOT SET>')}")
print(f"state             : {key_data.get('state', '<NOT SET>')}")
print(f"revocationReason  : {key_data.get('revocationReason', '<NOT SET>')}")
print(f"revocationMessage : {key_data.get('revocationMessage', '<NOT SET>')}")

print("\n=== What Terraform config specified ===")
print("revocation_reason  = 'KeyCompromise'")
print("revocation_message = 'test-repro-message-TFIN-286'")

print("\n=== Analysis ===")
reason_in_cm = key_data.get('revocationReason', '<NOT SET>')
message_in_cm = key_data.get('revocationMessage', '<NOT SET>')

if reason_in_cm == "test-repro-message-TFIN-286" and message_in_cm == "KeyCompromise":
    print("BUG CONFIRMED: revocationReason and revocationMessage are SWAPPED in CM!")
    print(f"  CM revocationReason  = '{reason_in_cm}'  (should be 'KeyCompromise')")
    print(f"  CM revocationMessage = '{message_in_cm}'  (should be 'test-repro-message-TFIN-286')")
elif reason_in_cm == "KeyCompromise" and message_in_cm == "test-repro-message-TFIN-286":
    print("NO BUG: values are correctly stored.")
else:
    print(f"UNEXPECTED: reason={reason_in_cm!r}, message={message_in_cm!r}")

File: .repro/check_api.go
package main

import (
	"bytes"
	"crypto/tls"
	"encoding/json"
	"fmt"
	"io"
	"net/http"
	"os"
)

const (
	cmHost = "https://ec2-44-205-177-172.compute-1.amazonaws.com"
	keyID  = "f11dd172f3ee4632ad55bb4834bcb95a4a73384647474fa3993bbab4862c7849"
)

func main() {
	tr := &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: true}}
	client := &http.Client{Transport: tr}

	// Get token
	authBody := `{"grant_type":"password","username":"admin","password":"Koala2.20","domain":"root","auth_domain":"root"}`
	req, _ := http.NewRequest("POST", cmHost+"/api/v1/auth/tokens", bytes.NewBufferString(authBody))
	req.Header.Set("Content-Type", "application/json")
	resp, err := client.Do(req)
	if err != nil {
		fmt.Fprintln(os.Stderr, "auth error:", err)
		os.Exit(1)
	}
	defer resp.Body.Close()
	body, _ := io.ReadAll(resp.Body)
	var tokenResp map[string]interface{}
	json.Unmarshal(body, &tokenResp)
	token, _ := tokenResp["jwt"].(string)
	fmt.Printf("Token obtained: %s...\n", token[:min(20, len(token))])

	// Query the key
	req2, _ := http.NewRequest("GET", cmHost+"/api/v1/vault/keys2/"+keyID, nil)
	req2.Header.Set("Authorization", "Bearer "+token)
	resp2, err := client.Do(req2)
	if err != nil {
		fmt.Fprintln(os.Stderr, "key query error:", err)
		os.Exit(1)
	}
	defer resp2.Body.Close()
	body2, _ := io.ReadAll(resp2.Body)
	var keyData map[string]interface{}
	json.Unmarshal(body2, &keyData)

	fmt.Println("\n=== CM API Response for key ===")
	fmt.Printf("id                : %v\n", keyData["id"])
	fmt.Printf("name              : %v\n", keyData["name"])
	fmt.Printf("state             : %v\n", keyData["state"])
	fmt.Printf("revocationReason  : %v\n", keyData["revocationReason"])
	fmt.Printf("revocationMessage : %v\n", keyData["revocationMessage"])

	fmt.Println("\n=== What Terraform config specified ===")
	fmt.Println("revocation_reason  = 'KeyCompromise'")
	fmt.Println("revocation_message = 'test-repro-message-TFIN-286'")

	reason := fmt.Sprintf("%v", keyData["revocationReason"])
	message := fmt.Sprintf("%v", keyData["revocationMessage"])

	fmt.Println("\n=== Analysis ===")
	if reason == "test-repro-message-TFIN-286" && message == "KeyCompromise" {
		fmt.Println("BUG CONFIRMED: revocationReason and revocationMessage are SWAPPED in CM!")
		fmt.Printf("  CM revocationReason  = '%s' (should be 'KeyCompromise')\n", reason)
		fmt.Printf("  CM revocationMessage = '%s' (should be 'test-repro-message-TFIN-286')\n", message)
	} else if reason == "KeyCompromise" && message == "test-repro-message-TFIN-286" {
		fmt.Println("NO BUG: values are correctly stored.")
	} else {
		fmt.Printf("UNEXPECTED: reason=%q, message=%q\n", reason, message)
	}
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}
